projects/core/src/product/services/coupon-search-page-meta.resolver.ts

```diff
@@ -0,0 +1,88 @@
+import { Injectable } from '@angular/core';
+import { ActivatedRoute } from '@angular/router';
+import { combineLatest, Observable } from 'rxjs';
+import { filter, map, switchMap } from 'rxjs/operators';
+import { AuthService } from '../../auth/facade/auth.service';
+import { BreadcrumbMeta, Page } from '../../cms/model/page.model';
+import {
+  PageBreadcrumbResolver,
+  PageMetaResolver,
+  PageTitleResolver,
+} from '../../cms/page';
+import { TranslationService } from '../../i18n/translation.service';
+import { PageType } from '../../model/cms.model';
+import { ProductSearchService } from '../../product/facade/product-search.service';
+import { SemanticPathService } from '../../routing/configurable-routes/url-translation/semantic-path.service';
+
+/**
+ * Resolves page meta data for the search result page, in case it's used
+ * to query coupons. This is done by adding a `couponcode` query parameter
+ * to the search page route.
+ *
+ * The page resolves an alternative page title and breadcrumb.
+ */
+@Injectable({
+  providedIn: 'root',
+})
+export class CouponSearchPageResolver extends PageMetaResolver
+  implements PageTitleResolver, PageBreadcrumbResolver {
+  protected total$: Observable<
+    number
+  > = this.productSearchService.getResults().pipe(
+    filter(data => !!data?.pagination),
+    map(results => results.pagination.totalResults)
+  );
+
+  constructor(
+    protected productSearchService: ProductSearchService,
+    protected translation: TranslationService,
+    protected authService: AuthService,
+    protected route: ActivatedRoute,
```

Comment:
Shouldn't we always use `getRouterState` instead of `ActivatedRoute`? There were some reported issues from the community that sometimes it wasn't working properly.

Reply:
I'm not aware of that issue. In this context I can only imagine that this happens if the page context is changing multiple times throughout the routing lifecycle (diff route events). If that's the case, it would not cause an issue, it would simply not score the resolver with enough score to be identified as the resolver for the coupon search result. 

If that would happen and it's causing issue, I rather like to fix the root issue. The score function is synchronise and won't allow to interact with the router state.